### ISSUE TITLE
<ref> inside <listRef>

### ISSUE THREAD

**Issue by scstanley7** [TEIC/Stylesheets#286]
Over the course of addressing issues [#1667](https://github.com/TEIC/TEI/issues/1667) and [#1674](https://github.com/TEIC/TEI/issues/1674) (in TEI), @sydb and I noticed that the component parts of [#1449](https://github.com/TEIC/TEI/issues/1449) (allowing `<ref>` inside `<listRef>`) were not fully addressed. Elli mentioned in a comment that the [ODD stylesheets would need to be changed](https://github.com/TEIC/TEI/issues/1449#issuecomment-214789475), but it appear that that hasn't happened.

Basically, `<ref>` was added to the content model of `<listRef>`, but the ODD stylesheets do not process `listRef/ref` in any way (they only address `<ptr>`). We either need to update this or introduce a Schematron warning when `<ref>` occurs inside `<listRef>` in an ODD.

**Comment by sydb** [comment]
If we choose to update the Stylesheets, looks to me at quick glance like a minimally invasive fix would be to change the `@match` on odd2odd.xsl:2058 to be `tei:ptr | tei:listRef/tei:ref`, and to change line 2076 to use the content of `<ref>` (instead of `<head>`). Better fixes exist, of course.

And this does **not** address processing of `<listRef>` in odd2lite.

**Comment by ebeshero** [comment]
@sydb I think you mean odd2odd.xsl:20**6**8, and I've got a nice list of other places to take a look at on Thursday for this one...

**Comment by ebeshero** [comment]
My proposed solution is in this Pull Request: https://github.com/TEIC/Stylesheets/pull/292
It's for odd2odd, but doesn't tackle odd2lite, which I'm hoping we can discuss more!

**Comment by ebeshero** [comment]
Note that our proposed solution (amended during Stylesheets Co-op meeting of 2 Nov 2017) does not account for nested `ref` elements inside `listRef/ref`.

**Comment by ebeshero** [comment]
Closing this ticket, and opening a new one to update odd2lite.
